enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,202 @@
+
+1. **Simplified Monitoring Architecture:** By replacing Prometheus Adapter with Metrics Server, we can reduce the complexity and maintenance overhead of the monitoring system in OpenShift since `metrics-server` does not rely on Prometheus.
+
+1. **Lightweight Solution:** Deploying and configuring Prometheus Adapter requires additional effort compared to Metrics Server. Prometheus Adapter is tightly coupled to Prometheus and this creates a bottleneck for allowing monitoring as  optional feature. See [MON-3152](https://issues.redhat.com/browse/MON-3152).
+
+
+### User Stories
+
+#### Story 1
+
+* Inorder to provide [optional built-in monitoring](https://issues.redhat.com/browse/MON-3152) feature we need to provide a way for customers to do HPA other than through Prometheus based Prometheus Adapter.
+
+#### Story 2
+
+* As a cluster admin, I want a more lightweight and efficient solution for monitoring resource metrics in my cluster, so that I can optimize cluster performance and resource allocation.
+
+#### Story 3
+
+* As a cluster admin, I want a simplified monitoring architecture in OpenShift, so that I can reduce the complexity and maintenance overhead of the monitoring system.
+
+### Goals
+
+* Replace Prometheus Adapter with Metrics Server while maintaining feature parity with the existing stack 
+* Ensure no performance regression occurs when replacing Prometheus Adapter with Metrics Server
+* Allow Prometheus Adapter to Metrics Server transition (and vice-versa) through a configuration or feature-flag.
+
+
+### Non-Goals
+
+* Allow optional prometheus stack
+
+## Proposal
+
+The proposal is to replace the Prometheus Adapter with the Metrics Server for monitoring resource metrics in OpenShift.
+
+### Workflow Description
+
+The workflow for using the Metrics Server in OpenShift for monitoring resource metrics would involve the following steps:
+
+1. User sets `enableMetricsServer: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config`
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: cluster-monitoring-config
+      namespace: openshift-monitoring
+    data:
+      config.yaml: |
+        enableMetricsServer: true 
+    ```
+
+1. Deploy and configure the metrics-server in the `openshift-monitoring` namespace
+1. Verify `metrics-server` resource API is functional by querying metrics-server endpoint (TBD)
+1. Update `APIService` resource to point to metrics-server API instead of `prometheus-adapter` API
+1. Delete/Scale Down `prometheus-adapter` deployment
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+  ```go
+  type ClusterMonitoringConfiguration struct {
+    // Enable MetricsServer
+    EnableMetricsServer bool `json:"enableMetricsServer,omitempty"`
```

Comment:
I would make it a bool pointer so that we can flip the default as we want.

```suggestion
    EnableMetricsServer *bool `json:"enableMetricsServer,omitempty"`
```